src/v/cloud_storage_clients/client_pool.cc

```diff
@@ -172,10 +172,15 @@ ss::future<> client_pool::stop() {
     // Wait until all leased objects are returned
     co_await _gate.close();
 
+    std::vector<ss::future<>> stops;
+    stops.reserve(_pool.size());
+
     for (auto& it : _pool) {
-        co_await it->stop();
+        stops.emplace_back(it->stop());
```

Comment:
in cases like this `pool::stop` should be marked `noexcept`. otherwise, if an exception is thrown during stopping (for example i think some of the stop implementations are interacting with gates) then all any collected futures in the stops vector get backgrounded.

Reply:
Yep. Great point. We should be more generous with our noexcept usage.